diff --git a/task1/datasets/__pycache__/__init__.cpython-310.pyc b/task1/datasets/__pycache__/__init__.cpython-310.pyc
index a90763b..a609e08 100644
Binary files a/task1/datasets/__pycache__/__init__.cpython-310.pyc and b/task1/datasets/__pycache__/__init__.cpython-310.pyc differ
diff --git a/task1/datasets/__pycache__/augment.cpython-310.pyc b/task1/datasets/__pycache__/augment.cpython-310.pyc
index de156a6..88795f2 100644
Binary files a/task1/datasets/__pycache__/augment.cpython-310.pyc and b/task1/datasets/__pycache__/augment.cpython-310.pyc differ
diff --git a/task1/datasets/__pycache__/datasets.cpython-310.pyc b/task1/datasets/__pycache__/datasets.cpython-310.pyc
index 38f05c7..5624144 100644
Binary files a/task1/datasets/__pycache__/datasets.cpython-310.pyc and b/task1/datasets/__pycache__/datasets.cpython-310.pyc differ
diff --git a/task1/datasets/__pycache__/preprocess.cpython-310.pyc b/task1/datasets/__pycache__/preprocess.cpython-310.pyc
index 364797a..841501c 100644
Binary files a/task1/datasets/__pycache__/preprocess.cpython-310.pyc and b/task1/datasets/__pycache__/preprocess.cpython-310.pyc differ
diff --git a/task1/networks/__pycache__/__init__.cpython-310.pyc b/task1/networks/__pycache__/__init__.cpython-310.pyc
index bf917b3..7e2a5f8 100644
Binary files a/task1/networks/__pycache__/__init__.cpython-310.pyc and b/task1/networks/__pycache__/__init__.cpython-310.pyc differ
diff --git a/task1/networks/__pycache__/networks.cpython-310.pyc b/task1/networks/__pycache__/networks.cpython-310.pyc
index 08dfa54..6425654 100644
Binary files a/task1/networks/__pycache__/networks.cpython-310.pyc and b/task1/networks/__pycache__/networks.cpython-310.pyc differ
diff --git a/task1/networks/__pycache__/resnet_emo.cpython-310.pyc b/task1/networks/__pycache__/resnet_emo.cpython-310.pyc
index a0aff47..0acef73 100644
Binary files a/task1/networks/__pycache__/resnet_emo.cpython-310.pyc and b/task1/networks/__pycache__/resnet_emo.cpython-310.pyc differ
diff --git a/task1/train.py b/task1/train.py
index 529606c..8236cf4 100644
--- a/task1/train.py
+++ b/task1/train.py
@@ -53,9 +53,9 @@ from torch.utils.data import DataLoader
 import torch.utils.data.distributed
 
 import sys
-import subprocess
-subprocess.check_call(["sudo", sys.executable, "-m", "pip", "install", "python-dotenv"])
-subprocess.check_call(["sudo", sys.executable, "-m", "pip", "install", "wandb"])
+import subprocess as sp
+sp.check_call(["sudo", sys.executable, "-m", "pip", "install", "python-dotenv"])
+sp.check_call(["sudo", sys.executable, "-m", "pip", "install", "wandb"])
 import wandb
 from dotenv import load_dotenv
 
@@ -93,7 +93,7 @@ def main(config, do_eval, save_path):
     # -- get train dataset
     df_train = pd.read_csv(f'{DATA_PATH}/info_etri20_emotion_train.csv')
     train_dataset = Datasets()(df_train, config['img_size'], base_path=f'{DATA_PATH}/Train/')
-    train_dataset = Augment(config['augment'])(train_dataset)
+    #train_dataset = Augment(config['augment'])(train_dataset)
     train_dataset = Preprocess(config['preprocess'])(train_dataset)
     train_dataloader = DataLoader(
         train_dataset, batch_size=config['batch_size'], shuffle=True, num_workers=0)
@@ -190,6 +190,7 @@ def main(config, do_eval, save_path):
         if ((epoch + 1) % 20 == 0):
             torch.save(net.state_dict(), save_path + '/model_' + str(epoch + 1) + '.pkl')
     torch.save(net.state_dict(), save_path + '/model_' + str(epoch + 1) + '.pkl')
+    wandb.finish()
 
 def acc(self, logit, label) :
     """
diff --git a/wandb-use.pdf b/wandb-use.pdf
new file mode 100644
index 0000000..e732316
Binary files /dev/null and b/wandb-use.pdf differ
